### ISSUE TITLE
[rllib] Fix YAML pretty printing

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#1540]
Weird floats are printed from the pretty printing when running `./train.py -f tuned_examples/pong-dqn.yaml`
```
TrainingResult for DQN_PongDeterministic-v4_0:
  date: 2018-02-14_02-33-57
  episode_len_mean: 889.16
  episode_reward_mean: -20.48
  episodes_total: !!python/long '26'
  experiment_id: 21eb05251cd746c9802ac8d154a59994
  hostname: ip-172-31-66-3
  info:
    grad_time_ms: 8.038
    max_exploration: 0.88615
    min_exploration: 0.88615
    num_target_updates: !!python/long '23'
    opt_peak_throughput: 3981.151
    opt_samples: 32.0
    sample_time_ms: 10.988
    update_time_ms: 0.001
  pid: !!python/long '14587'
  time_this_iter_s: 4.731676816940308
```

Also, tune yaml docs are pretty underspecified.

**Comment by ericl** [comment]
safe_dump in yaml fixes this. Seems like in python2 pyyaml defaults to this weird thing with plain dump().

**Comment by richardliaw** [comment]
Fixed in #1655
